TheMET/TheMET/Network/MetAPI.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  MetAPI.swift
+//  TheMET
+//
+//  Created by Анна Ситникова on 25/04/2023.
+//
+
+import Foundation
+
+class MetAPI{
+    
+    private var networkManager: NetworkManager
+    
+    init(networkManager: NetworkManager) {
+        self.networkManager = networkManager
+    }
+    
+    func objects(metadataDate: Date? = nil, departmentIds: [Int] = [], completion: @escaping (ObjectsResponse?) -> Void) {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd"
+        let dateString: String?
+        if let date = metadataDate {
+            dateString = dateFormatter.string(from: date)
+        } else {
+            dateString = nil
+        }
+        let idArrayString: String?
+        if !departmentIds.isEmpty {
+            idArrayString = departmentIds.map {String($0)}.joined(separator: "|")
+        } else {
+            idArrayString = nil
+        }
+        let urlString: String = "https://collectionapi.metmuseum.org/public/collection/v1/objects"
+        var parameters: [String : String]
+        if let date =  dateString,
+           let arrayString = idArrayString {
+           parameters = ["metadataDate" : date, "departmentIds" : arrayString]
+        } else if let date =  dateString {
+            parameters = ["metadataDate" : date]
+        } else if let arrayString = idArrayString {
+            parameters = ["departmentIds" : arrayString]
+        } else {
+            parameters = [ : ]
+        }
```

Comment:
You don't need to check the combinations of all the parameters with `if else` chain. Imagine how hard it would be to manage 3 or 4 parameters.

Instead, create an empty dictionary. Then put a `date` and `idArrayString` to the dictionary if needed.